ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -169,20 +170,31 @@ private[server] object ServerHelpers {
   private[internal] def postProcessResponse[F[_]: Timer: Monad](
       req: Request[F],
       resp: Response[F]): F[Response[F]] = {
-    val reqHasClose = req.headers.exists {
-      // We know this is raw because we have not parsed any headers in the underlying alg.
-      // If Headers are being parsed into processed for in ParseHeaders this is incorrect.
-      case Header.Raw(name, values) => name == connectionCi && values.contains(closeCi.value)
-      case _ => false
-    }
     val connection: Connection =
-      if (reqHasClose) close
-      else keepAlive
+      if (isKeepAlive(req.httpVersion, req.headers)) keepAlive
+      else close
     for {
       date <- HttpDate.current[F].map(Date(_))
     } yield resp.withHeaders(Headers.of(date, connection) ++ resp.headers)
   }
 
+  private[internal] def isKeepAlive(httpVersion: HttpVersion, headers: Headers): Boolean = {
+    // We know this is raw because we have not parsed any headers in the underlying alg.
+    // If Headers are being parsed into processed for in ParseHeaders this is incorrect.
+    def hasConnection(expected: String): Boolean =
+      headers.exists {
+        case Header.Raw(name, value) =>
+          name == connectionCi && value.toLowerCase.contains(expected)
```

Comment:
Should always use a `Locale` on `toLowerCase`.  Most people use `Locale.US`, but I use `Locale.ROOT`.

Reply:
This will generate false positives if `keep-alive` or `close` are substrings of a different token, but I suppose those people are just asking for trouble.